azimuth/types/model_performance.py

```diff
@@ -30,18 +34,46 @@ class UtteranceCountPerFilterValue(AliasModel):
     filter_value: str = Field(..., title="Filter value")
 
 
-class UtteranceCountPerFilter(AliasModel):
-    prediction: Optional[List[UtteranceCountPerFilterValue]] = Field(
-        ..., title="Prediction", nullable=True
+if typing.TYPE_CHECKING:
+
+    class GenericAliasModel(AliasModel, Generic[T]):
+        pass
+
+    class GenericAliasModelPipeline(AliasModel, Generic[T]):
+        pass
+
+    ValuePerDatasetSmartTag = GenericAliasModel
+    ValuePerPipelineSmartTag = GenericAliasModelPipeline
+
+else:
+    ValuePerDatasetSmartTag = AliasModel.with_fields(
+        "ValuePerDatasetSmartTag",
+        GenericModel,
+        Generic[T],
+        **{k.value: (List[T], Field(..., title=k.value)) for k in SMART_TAG_FAMILY_DATASET}
```

Comment:
If the `title` is the same as the field name, is it really useful? Can we just omit the `Field` altogether? It's not important, I'm just curious.
```suggestion
        **{k.value: List[T] for k in SMART_TAG_FAMILY_DATASET}
```